### PR TITLE
Auth: Enable more complete credential chain for SigV4 default SDK auth option

### LINKED ISSUE #28931: Sigv4 returns 403 when running from EKS with IAM role for service Accounts
We are running Grafana from an AWS EKS cluster (v1.18) and connecting to AWS Elastic Search service version 7.7.

We are using Service accounts and IAM roles in our pod (https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html)

At the moment we are successfully connecting to ElasticSearch with aws-sigv4-proxy as a sidecar container using the proxy URL as URL:

image

When we change to use the ElasticSearch endpoint and enable Sigv4:

image

We get 403:

image

Not relevant error other than request completed with 403 response:

`info msg="Request Completed" logger=context userId=39 orgId=1 uname=----- method=GET path=------- status=403 remote_addr=------- time_ms=3175 size=141 referer=https://grafana.------`

**What happened**:

Can't use Sigv4 in our Elastic Search Data Source when running Grafana from Eks with IAM roles for Service Accounts

**What you expected to happen**:

Able to use new Sigv4 feature and remove our aws-sigv4-proxy

**How to reproduce it (as minimally and precisely as possible)**:

Add Grafana to an EKS cluster with a service account and IAM role and try to connect to ES using Sigv4.

For clarity: and in case it helps: https://docs.aws.amazon.com/eks/latest/userguide/specify-service-account-role.html

### PR DESCRIPTION
**Pull request by wbrowne**
**Which issue(s) this PR fixes**:

Fixes #28931

**Special notes:** 
`defaults.CredChain` doesn't include the IAM roles for Service Accounts credential provider but the default cred chain in `session.NewSession()` does. See here: https://github.com/aws/aws-sdk-go/blob/d6d63adc1c143626733f1ab3d575b6f777357d04/aws/session/credentials.go#L18

### CODE REVIEW

**Review by marefr — APPROVED** [review]
LGTM 

Is this something needed for cloudwatch as well?

**Review by papagian — APPROVED** [review]
LGTM!

File: pkg/models/sigv4.go [hunk 1]
```diff
@@ -90,9 +89,28 @@ func (m *SigV4Middleware) signRequest(req *http.Request) (http.Header, error) {
...
+		c = credentials.NewStaticCredentials(m.Config.AccessKey, m.Config.SecretKey, "")
+	case Credentials:
+		c = credentials.NewSharedCredentials("", m.Config.Profile)
+	}
+
+	// passing nil credentials will force AWS to allow more complete credential chain vs the explicit default
+	if c == nil {
+		s, err := session.NewSession(&aws.Config{
+			Region:      aws.String(m.Config.Region),
+			Credentials: nil},
```
Comment (aknuds1) [inline pkg/models/sigv4.go:108]: There's no need to provide the default value.
```suggestion
		},
```

**Review by aknuds1 — APPROVED** [review]
Looks good AFAICT, with some nits.

**Comment by wbrowne** [comment]
@marefr SigV4 and CloudWatch should behave consistently now with this change (so CloudWatch was always working as expected AFAICT).  I've tested it out locally with the default SDK option and  I'm able to reach [here](https://github.com/aws/aws-sdk-go/blob/d6d63adc1c143626733f1ab3d575b6f777357d04/aws/session/credentials.go#L18), which will look for the extended default options like the one mentioned in #29017. I think we would need to investigate further and try it out on the version specified and compare with 7.3.X since the credentials chain might have changed between versions.

**Review by marefr — APPROVED** [review]
LGTM

**Pull request merged**